src/av2/structures/ndgrid.py

```diff
@@ -142,16 +145,23 @@ def points_to_bev_img(
 
         points_xy = points[..., :2].copy()  # Prevent modifying input.
         indices = self.transform_to_grid_coordinates(points_xy)
-        indices_int, _ = crop_points(indices, lower_bound_inclusive=(0.0, 0.0), upper_bound_exclusive=self.dims)
+        indices_int, valid_mask = crop_points(
+            indices,
+            lower_bound_inclusive=(0.0, 0.0),
+            upper_bound_exclusive=self.dims,
+        )
 
         # Construct uv coordinates.
         H, W = (self.dims[0], self.dims[1])
         uv = indices_int[..., :2]
 
-        C = len(color)
-        shape = (H, W, C)
+        shape = (H, W, 3)
         img: NDArrayByte = np.zeros(shape, dtype=np.uint8)
 
-        colors: NDArrayByte = np.array([color for _ in range(len(points_xy))], dtype=np.uint8)
+        if isinstance(color, tuple):
+            colors: NDArrayByte = np.array([color for _ in range(len(points_xy))], dtype=np.uint8)
+        else:
+            # Use lidar intensity
+            colors = np.array([[gray_val] * 3 for gray_val in color[valid_mask]], dtype=np.uint8)
```

Comment:
Could we call `np.full` here? I think it may be cleaner.

Reply:
Yeah!